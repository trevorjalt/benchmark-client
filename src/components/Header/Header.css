.Header {
    height: 20vh;
    display: flex;
    align-items: center;
    justify-content: space-between;;
    background-color: #FFD23F;
    padding: 1vw 2vw 1vw 2vw;
    margin-bottom: 4vh;
    text-decoration: none;
}

.Header__text {
    font-weight: bold;
    display: flex;
    flex-direction: row;
    align-items: center;
}

h1 {
    display: flex;
    align-items: center;
}

.logo {
    font-size: 6vh;
    text-decoration: none;
    color: black;
}

.logoIcon {
    height: 5.5vh;
    margin: 0 2vw 0 2vw;
}

.navIcon {
    height: 4vh;
    margin: 2vw
}

.navItem {
    display: flex;
    align-items: center;
    text-decoration: none;
    color: black;
}

.navItem__text {
    display: none;
}

.navItem__text2 {
    display: none;
}


@media only screen and (min-width: 800px) {
    .Header__not-logged-in, .Header__logged-in {
        display: flex;
        align-items: center;
    }

    .logo {
        font-size: 8vh;
    }
    
    .logoIcon {
        height: 10vh;
    }

    .navIcon {
        height: 6vh;
        margin: 0 1vw 0 2vw;
    }
    
    .navItem__text {
        display: block;
    }

    .navItem__text2 {
        display: block;
        margin-right: 2vw;
    }
}