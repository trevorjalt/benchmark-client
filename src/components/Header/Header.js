import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import IdleService from '../../services/idle-service'
import WorkoutContext from '../../contexts/WorkoutContext'
import Logo from './images/benchmark-logo.png'
import LoginIcon from './images/login-icon.png'
import LogoutIcon from './images/logout-icon.png'
import MyWorkoutsIcon from './images/my-workouts-icon.png'
import NewWorkoutIcon from './images/new-workout-icon.png'
import RegisterIcon from './images/register-icon.png'
import './Header.css'

export default class Header extends Component {
    static contextType = WorkoutContext

    handleLogoutClick = () => {
        TokenService.clearAuthToken()
        TokenService.clearCallbackBeforeExpiry()
        IdleService.unRegisterIdleResets()
        this.context.setIsLoggedIn(false)
    }

    renderLogoutLink() {
        return (
            <div className='Header__logged-in'>
                <Link
                    to='/myworkouts'
                    className='navItem'
                >
                    <img src={MyWorkoutsIcon} alt='My workouts link logo' className='navIcon'></img>
                    <p className='navItem__text'>myWorkouts</p>
                </Link>
                <Link
                    to='/newworkout'
                    className='navItem'
                >
                    <img src={NewWorkoutIcon} alt='New workout link logo' className='navIcon'></img>
                    <p className='navItem__text'>newWorkout</p>
                </Link>
                <Link
                    to='/'
                    className='navItem'
                    onClick={this.handleLogoutClick}
                >
                    <img src={LogoutIcon} alt='Logout link logo' className='navIcon'></img>
                    <p className='navItem__text2'>Logout</p>
                </Link>
            </div>
        )
    }

    renderLoginLink() {
        return (
            <div className='Header__not-logged-in'>
                <Link
                    to='/register'
                    className='navItem'
                >
                    <img src={RegisterIcon} alt='Register link logo' className='navIcon'></img>
                    <p className='navItem__text'>Register</p>
                </Link>
                <Link
                    to='/login'
                    className='navItem'
                >
                    <img src={LoginIcon} alt='Login link logo' className='navIcon'></img>
                    <p className='navItem__text2'>Login</p>
                </Link>
            </div>
        )
    }

    render() {
        return (
            <nav className='Header'>
                <span className='Header__text'>
                    <img src={Logo} alt='benchmark logo' className='logoIcon' />
                    <Link to='/' className='logo'>
                        benchMark
                    </Link>
                </span>                           
                {TokenService.hasAuthToken()
                    ? this.renderLogoutLink()
                    : this.renderLoginLink()}                 
            </nav>        
        )
    }
}