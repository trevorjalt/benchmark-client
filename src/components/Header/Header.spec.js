import React from 'react'
import Header from './Header'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`Header Component`, () => {
    it (`renders without crashing`, () => {
        const wrapper = 
            shallow (
                <Header />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})