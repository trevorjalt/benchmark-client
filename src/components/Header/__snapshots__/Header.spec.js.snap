// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders without crashing 1`] = `
<nav
  className="Header"
>
  <span
    className="Header__text"
  >
    <img
      alt="benchmark logo"
      className="logoIcon"
      src="benchmark-logo.png"
    />
    <Link
      className="logo"
      to="/"
    >
      benchMark
    </Link>
  </span>
  <div
    className="Header__not-logged-in"
  >
    <Link
      className="navItem"
      to="/register"
    >
      <img
        alt="Register link logo"
        className="navIcon"
        src="register-icon.png"
      />
      <p
        className="navItem__text"
      >
        Register
      </p>
    </Link>
    <Link
      className="navItem"
      to="/login"
    >
      <img
        alt="Login link logo"
        className="navIcon"
        src="login-icon.png"
      />
      <p
        className="navItem__text2"
      >
        Login
      </p>
    </Link>
  </div>
</nav>
`;
